spec/post_index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 0, bio: 'A beautiful bio')
+    @post = Post.create(
+      title: 'Post title',
+      text: 'Post text',
+      comments_counter: 0,
+      likes_counter: 0,
+      author: @user
+    )
+  end
+
+  describe 'Post Index page' do
+    before(:each) do
+      visit "/users/#{@user.id}/posts"
+    end
+    it 'displays user profile information' do
+      expect(page).to have_css("img[src*='#{@user.photo}']")
+      expect(page).to have_content(@user.name)
+      expect(page).to have_content("Number of posts: #{@user.post_counter}")
+    end
+
+    it 'displays each post with relevant information' do
+      @posts = @user.posts
+
+      @posts.each do |post|
+        expect(page).to have_content(post.title)
+        expect(page).to have_content(post.text[0..49])
+        expect(page).to have_content("Comments: #{post.comments_counter}")
+        expect(page).to have_content("Likes: #{post.likes_counter}")
+        # expect(page).to have_link("#{post.title}", href: "users/1/posts/1")
```

Comment:
- [ ] Kindly remove commented code. They can make your code hard to read.

Reply:
@julie-ify done on this change